### PR TITLE
Feature/add multi address ui

### PR DESCRIPTION
**Pull request by Firefight-Rhizome**
image

image

### CODE REVIEW

File: MetrICX-UI/src/app/addressModal/addressModal.page.ts [hunk 1]
```diff
@@ -16,19 +15,50 @@ export class AddressModalPage {
...
+  ) { }
 
+  async ionViewWillEnter() {
+     if(this.navParams.get('key')) {
+       this.key = this.navParams.get('key');
+       var settings = await this.settingsService.get();
+       if (settings.addresses_v2) {
+         this.isUpdate = true;
+          this.Address = JSON.parse(JSON.stringify(settings.addresses_v2));
+          this.address = settings.addresses_v2[this.key].address;
```
Comment (salvation-al) [inline MetrICX-UI/src/app/addressModal/addressModal.page.ts]: A bit confusing this.address vs this.Address, can be fixed later as well

File: MetrICX-UI/src/app/services/settings/settings.service.ts [hunk 2]
```diff
@@ -70,15 +70,35 @@ export class SettingsService {
   public getActiveAddress() : Address {
     return this.deviceSettings.addresses_v2.p0;
   }
+public async updateAndSave(key: string, address: string, nickName: string, enablePushDeposits: boolean) : Promise<boolean> {
```
Comment (salvation-al) [inline MetrICX-UI/src/app/services/settings/settings.service.ts]: In this file, I feel there should be something somewhere that will copy the value of enablePushDeposits into any / all existing addresses already configured

File: MetrICX-UI/src/app/addressModal/addressModal.page.ts [hunk 3]
```diff
@@ -16,19 +15,50 @@ export class AddressModalPage {
...
+    public navParams: NavParams   
+  ) { }
 
+  async ionViewWillEnter() {
+     if(this.navParams.get('key')) {
+       this.key = this.navParams.get('key');
+       var settings = await this.settingsService.get();
+       if (settings.addresses_v2) {
+         this.isUpdate = true;
+          this.Address = JSON.parse(JSON.stringify(settings.addresses_v2));
```
Comment (salvation-al) [inline MetrICX-UI/src/app/addressModal/addressModal.page.ts]: Address vs address, a bit confusing

File: MetrICX-UI/src/app/addressModal/addressModal.page.ts (hunk 1 above)
Comment (Firefight-Rhizome) [reply MetrICX-UI/src/app/addressModal/addressModal.page.ts]: yes I tried to make the attribute .Address but you said the backend would screw up

File: MetrICX-UI/src/app/services/settings/settings.service.ts (hunk 2 above)
Comment (Firefight-Rhizome) [reply MetrICX-UI/src/app/services/settings/settings.service.ts]: Yeh, the existing address already configured ill work on in a separate branch

**Pull request merged**